From bde37f5a75283d53043a0505359e8a8612448d0d Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Mon, 23 May 2011 11:33:47 +0300
Subject: [PATCH 54/75] wl12xx: set band and channel on cmd_role_start_dev()

Their value isn't really important (as we are going to ROC anyway),
but the default 0 value was invalid.
---
 drivers/net/wireless/wl12xx/cmd.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/net/wireless/wl12xx/cmd.c b/drivers/net/wireless/wl12xx/cmd.c
index a47490f..7c5ce77 100644
--- a/drivers/net/wireless/wl12xx/cmd.c
+++ b/drivers/net/wireless/wl12xx/cmd.c
@@ -480,6 +480,10 @@ int wl1271_cmd_role_start_dev(struct wl1271 *wl)
 	wl1271_debug(DEBUG_CMD, "cmd role start dev %d", wl->role_id);
 
 	cmd->role_id = wl->dev_role_id;
+	if (wl->band == IEEE80211_BAND_5GHZ)
+		cmd->band |= WL1271_BAND_5GHZ;
+	cmd->channel = wl->channel;
+
 	if (wl->dev_hlid == WL1271_INVALID_LINK_ID) {
 		ret = wl1271_allocate_link(wl, &wl->dev_hlid);
 		if (ret)
-- 
1.7.0.4

